uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -254,6 +252,22 @@ define([
       };
 
       /**
+       * Checks if a leave request has been updated since opening the modal
+       *
+       * @return {Boolean}
+       */
+      this.hasRequestUpdated = function () {
+        // angular copy is necessary to remove the $$hashkey property from
+        // all nested objects
+        var updatedRequestAttributes = angular.copy(this.request.attributes());
+        // fileUploader property deleted because it will not be used
+        // in object comparison
+        delete updatedRequestAttributes.fileUploader;
+
+        return !_.isEqual(initialLeaveRequestAttributes, updatedRequestAttributes);
```

Comment:
do you think [`angular.equals`](https://docs.angularjs.org/api/ng/function/angular.equals) can solve the problem with `$$hashkey`?